src/login/logind-session-device.c

```diff
@@ -387,14 +407,11 @@ int session_device_new(Session *s, dev_t dev, SessionDevice **out) {
         }
         sd->fd = r;
 
-        LIST_PREPEND(sd_by_device, sd->device->session_devices, sd);
-
         *out = sd;
         return 0;
 
 error:
-        hashmap_remove(s->devices, &sd->dev);
-        free(sd->node);
+        session_device_deinit(sd);
         free(sd);
```

Comment:
I think I should remove the commit that introduced session_device_{init,deinit}() functions. They were initially introduced because the device initialization was needed by 2 different paths.

But since then, the patchset has been reworked and it's no longer true.